core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -496,8 +497,9 @@ public void run() {
          throw new CacheException("Interrupted while waiting for the caches to start");
       }
 
-      if (exception.get() != null) {
-         throw exception.get();
+      RuntimeException runtimeException = exception.get();
```

Comment:
Intellij inspections keep telling me (wrongly) that the exception is likely lost and not rethrown, so I fixed it.